ONSdigital/response-operations-ui
# Review warnings in resops tests

# Motivation and Context
The unit tests were throwing a lot of warnings when run.  This PR fixes them.

# What has changed
The unit tests were encoding/decoding the jwt differently from how it does it when the service is running (using symmetric keys instead of asymmetric ones used by the service).  The keys are configured in the config.py file and now we can restrict the algorithm used by the jwt module.

Additionally, there were deprecation warnings in the uaa tests.  The way we were testing the exception being raised wasn't working and I had to change the test itself.  Because the result of the function isn't a raised exception but instead is `None` and a logger.exception line, I had to change the thing that's being tested.  We maybe could(should?) change the `request_uaa_public_key` function to `raise` after logging instead of returning `None` on error... but that's a PR for a different time.  Feel free to raise a card if you agree with me whilst reviewing it.

# How to test?
Run the unit tests - Should pass with 0 warnings
Run the app - should work just as before as no functionality should've changed.

# Links
https://trello.com/c/P9hzNOLu/802-spike-review-warnings-in-resops-tests-1-day
